database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -25,7 +25,7 @@
  * This class implements an array-like collection on top of a Firebase location.
  */
 class FirebaseArray implements ChildEventListener {
```

Comment:
In terms of SOLID programming, shouldn't at least the class itself be at least public for it to be open to extension? Building another adapter as an add-on to FirebaseUI using FirebaseArray wouldn't be possible with this configuration.